File: 프로그래머스/Lv1/음양 더하기.js
// 내 풀이
function solution(absolutes, signs) {
    let answer = 0;
    
    for(let i=0; i<signs.length; i++){
        if(signs[i]){
            answer += absolutes[i];
        }else{
            answer -= absolutes[i];
        }
    }
    return answer;
}

// 다른 사람 풀이
function solution2(absolutes, signs){
    return absolutes.reduce((a, c, i) => a + c * (signs[i] ? 1 : -1), 0);
}

console.log(solution([4,7,12], [true,false,true]));
console.log(solution2([1,2,3], [false,false,true]));

File: 프로그래머스/Lv1/음양 더하기.py
# 내 풀이
def solution(absolutes, signs):
    return sum([x * (1 if y else -1) for x, y in zip(absolutes, signs)])

# 다른 사람 풀이
def solution2(absolutes, signs):
    return sum(absolutes if sign else -absolutes for absolutes, sign in zip(absolutes, signs))

print(solution([4,7,12], [True,False,True]));
print(solution2([1,2,3], [False,False,True]));